content/5_Plat/Binary_Jump.mdx

```diff
@@ -307,6 +307,79 @@ int main() {
 
 </LanguageSection>
 
+### Alternate Solution II (Company Queries II)
+
+We can also find the LCA of two nodes using the Tarjan's Offline LCA algorithm. By taking advantage of the DFS traversal, we 
+can precompute the answers to the queries through forming subtrees and calculating the common parent with a similar structure 
+as Disjoint-Set Union.  
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+CodeSnip{Benq Template}
+
+int N, M;
+vpi adj[200001], q[200001];
```

Comment:
I think these can all be 200000 instead of 200001? Use a `const int` to represent the max size instead of copying and pasting the same number.

Reply:
> I think these can all be 200000 instead of 200001? Use a `const int` to represent the max size instead of copying and pasting the same number.

I used 1-indexing so I think it is 200001? Or I could convert it to 0-indexed if that's better.

Reply:
nvm I just got confused by some of the macros 😓